helper/pluginutil/runner.go

```diff
@@ -70,6 +71,7 @@ func (r *PluginRunner) runCommon(ctx context.Context, wrapper RunnerUtil, plugin
 	if wrapper.MlockEnabled() {
 		cmd.Env = append(cmd.Env, fmt.Sprintf("%s=%s", PluginMlockEnabled, "true"))
 	}
+	cmd.Env = append(cmd.Env, fmt.Sprintf("%s=%s", PluginVaultVersionEnv, version.GetVersion().VersionNumber()))
```

Comment:
`version.GetVersion().VersionNumber()` would return "(version unknown)" if `if Version == "unknown" && VersionPrerelease == "unknown"` is satisfied. The check on https://github.com/hashicorp/vault/pull/3833/files#diff-cedffd7b22c2dfe1ab831e6c9f38e45dR19 would need to be changed to check for that instead.

Reply:
Is the "unknown" check below not sufficient?

Reply:
version.GetVersion().VersionNumber() would never return the string "unknown" (in returns "(version unknown)") so `verString != "unknown"` will always be true, even when the version is unknown.